controller/controller.go

```diff
@@ -285,17 +289,31 @@ func (f *Controller) bootWithError(ctx context.Context) error {
 		}
 
 		f.logger.LogCtx(ctx, "function", "bootWithError", "level", "debug", "message", "ensured custom resource definition exists")
-
-		// TODO collect metrics
 	}
 
-	f.logger.LogCtx(ctx, "function", "bootWithError", "level", "debug", "message", "booting informer")
-	err := f.informer.Boot(ctx)
-	if err != nil {
-		return microerror.Mask(err)
+	{
+		f.logger.LogCtx(ctx, "function", "bootWithError", "level", "debug", "message", "booting informer")
+
+		err := f.informer.Boot(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		f.logger.LogCtx(ctx, "function", "bootWithError", "level", "debug", "message", "booted informer")
 	}
 
-	f.logger.LogCtx(ctx, "function", "bootWithError", "level", "debug", "message", "booted informer")
+	go func() {
+		resetWait := f.informer.ResyncPeriod() * 2
+
+		for {
+			select {
+			case <-f.errorCollector:
+				controllerErrorGauge.Inc()
+			case <-time.After(resetWait):
+				controllerErrorGauge.Set(0)
+			}
+		}
+	}()
```

Comment:
> I think what you want to achieve can be solved by using counter and rate function

No it cannot, which is why I did this PR here. Talked this through already with Timo and Marcel. No Prometheus magic right now can do this. Everything we tried so far resulted in flapping alerts because we do not get a stable signal using Prometheus queries but only waves that signal and stop signalling. This is due to the reconciliation loops and how often we scrape.